reference/nonfungible-token.scilla

```diff
@@ -64,257 +71,472 @@ let makeErrorEvent =
+transition name()
+  msg_to_sender = { _tag : "nameCallBack"; _recipient : _sender; _amount : Uint128 0;
+                   name : name};
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* @dev: Get name of the NFTs *)
+transition symbol()
+  msg_to_sender = { _tag : "symbolCallBack"; _recipient : _sender; _amount : Uint128 0;
+                   symbol : symbol};
+  msgs = one_msg msg_to_sender;
+  send msgs
 end
 
-(* @dev:    Mint new tokens. Only contractOwner can mint. *)
-(* @param:  to      - Address of the token recipient      *)
-(* @param:  tokenId - ID of the new token minted          *)
-(* Returns error message CodeTokenExists if token exists. *)
-(* Revert transition if invalid recipient contract.       *)
-transition mint(to: ByStr20, tokenId: Uint256)
+(* @dev: Get approved_addr for token_id *)
+transition getApproved(token_id: Uint256)
+  some_token_approval <- token_approvals[token_id];
+  match some_token_approval with
+  | Some addr => 
+    msg_to_sender = { _tag : "getApprovedCallBack"; _recipient : _sender; _amount : Uint128 0; 
+                      approved_addr : addr};
+    msgs = one_msg msg_to_sender;
+    send msgs
+  | None => throw
+  end
+end
+
+(* @dev: Get the token_uri of a certain token_id *)
+transition getTokenURI(token_id: Uint256)
+  some_token_uri <- token_uris[token_id];
+  match some_token_uri with
+  | Some token_uri =>
+    msg_to_sender = { _tag : "getTokenURICallBack"; _recipient : _sender; _amount : Uint128 0; 
+                      token_uri : token_uri};
+    msgs = one_msg msg_to_sender;
+    send msgs
+  | None => throw
+  end
+end
+
+(* @dev: Check if a token_id is owned by a token_owner and get a Bool *)
+transition isOwner(token_id: Uint256, address: ByStr20)
+  some_token_owner <- token_owners[token_id];
+  is_owner_bool = 
+    match some_token_owner with
+    | Some addr => builtin eq addr address
+    | None => False
+    end;
+  msg_to_sender = { _tag : "isOwnerCallBack"; _recipient : _sender; _amount : Uint128 0;
+                    is_owner : is_owner_bool};
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* @dev: Check if address is operator for token_owner and get a Bool *)
+transition isApprovedForAll(token_owner: ByStr20, operator: ByStr20)
+  some_operator_approval <- operator_approvals[token_owner][operator];
+  is_operator =
+    match some_operator_approval with
+    | Some status => status
+    | None => False
+    end;
+  msg_to_sender = { _tag : "isApprovedForAllCallBack"; _recipient : _sender; _amount : Uint128 0; 
+                   is_operator : is_operator};
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Interface transitions *)
+
+(* @dev:    Add or remove approved minters. Only contract_owner can approve minters. *)
+(* @param:  minter      - Address of the minter to be approved or removed            *)
+(* Returns error event CodeNotAuthorised if _sender is not contract_owner.           *)
+transition configureMinter(minter: ByStr20)
```

Comment:
this transition looks very confusing to me. 
It's very error-prone. Is this from an existing contract standard? I can't find it on github https://github.com/0xjac/ERC777/blob/devel/contracts/ERC20Token.sol

Reply:
I needed a way to approve and remove `minters` role from the contract, so I've added this `configureMinter` transition.
Currently, `mint` and `burn` are both privileged transitions that can only be called by `minters`.
Any suggestion to make this less error-prone?